src/features/completions.zig

```diff
@@ -1366,13 +1335,27 @@ fn collectFieldAccessContainerNodes(
             const fn_proto_node = node_type.type.data.other; // this assumes that function types can only be Ast nodes
             var buf: [1]Ast.Node.Index = undefined;
             const full_fn_proto = node_type.handle.tree.fullFnProto(&buf, fn_proto_node).?;
+            if (dot_context.need_ret_type) { // => we need f()'s return type
+                const has_body = node_type.handle.tree.nodes.items(.tag)[node_type.type.data.other] == .fn_decl;
+                const body = node_type.handle.tree.nodes.items(.data)[node_type.type.data.other].rhs;
+                node_type = try analyser.resolveReturnType(full_fn_proto, node_type.handle, if (has_body) body else null) orelse return;
+                // `E!T`
+                if (node_type.type.data == .other and node_type.handle.tree.nodes.items(.tag)[node_type.type.data.other] == .error_union) {
+                    const node_data = node_type.handle.tree.nodes.items(.data)[node_type.type.data.other];
+                    if (node_data.rhs == 0) return;
+                    node_type = try analyser.resolveTypeOfNode(.{ .node = node_data.rhs, .handle = node_type.handle }) orelse return;
+                }
+                try node_type.getAllTypesWithHandlesArrayList(arena, types_with_handles);
+                return;
```

Comment:
This code snippet appears three times in your dot completion code. Perhaps its better move this logic into analysis.
The code duplication seems to have caused a mistake because you are doing `return` instead of `continue` on failure (you are in a for loop here).

I guess you are often looking for a function that does the following conversion:
```rust
match(T) {
    ?P -> P,
    E!P -> P,
    ?E!P -> P,
    else -> T,
}
```
Its like the reverse of type coercion.